tests/unit/async/Task.ts

```diff
@@ -372,6 +372,46 @@ function addPromiseTests(suite: any, Promise: any) {
 			Promise.all(iterable).then(dfd.callback(function (value: number[]) {
 				assert.notStrictEqual(value, iterable);
 			}));
+		},
+
+		'cancelable': {
+			'isIterable': function (this: any) {
+				let pending: any[] = [];
+
+				for (let i = 0; i < 3; i++) {
+					pending[i] = new Promise(function () {});
+				}
+
+				let tasks = Promise.all(pending);
+
+				tasks.cancel();
+
+				tasks.then(() => {
```

Comment:
This is literally what the tests are asserting: `assert.strictEqual(task.state, State.canceled, 'Task should have Canceled state');` where `task` is each item of `pending`.